internal/job/executor.go

```diff
@@ -521,7 +521,7 @@ func (e *Executor) applyEnvironmentChanges(changes hook.EnvChanges, redactors re
 	}
 	for k, v := range changes.Diff.Changed {
 		if _, ok := executorConfigEnvChanges[k]; ok {
-			e.shell.Commentf("%s is now %q", k, v)
+			e.shell.Commentf("%s was %q and is now %q", k, v.Old, v.New)
```

Comment:
`v` is an [`env.DiffPair`](https://github.com/buildkite/agent/blob/20f215b5bc69ec968d08cd0cabd49ddb50e0cf67/env/environment.go#L284-L287)